src/sys/linux_macos.rs

```diff
@@ -4,11 +4,26 @@ use std::mem;
 use std::os::unix::ffi::OsStrExt;
 use std::os::unix::io::BorrowedFd;
 use std::path::Path;
+use std::slice;
 
 use rustix::fs as rfs;
 
 use crate::util::allocate_loop;
 
+use std::os::raw::c_char;
+
+// Convert an `&mut [u8]` to an `&mut [c_char]`
+#[inline]
+fn as_list_buffer(buf: &mut [u8]) -> &mut [c_char] {
```

Comment:
nit: `as_c_buffer`?

Reply:
(I'm just not sure what "list buffer" means here).

Reply:
The buffer is for `listxattr`, I've renamed it accordingly.